### PR TITLE
10 - Zeroable module

### PR DESCRIPTION
**Pull request by TAdev0**

### CODE REVIEW

File: corelib/src/zeroable.cairo [hunk 1]
```diff
@@ -1,23 +1,40 @@
...
+//!
+//! ```
+//! use core::num::traits::Zero;
+//!
+//! assert!(Zero::<u32>::zero() == 0);
+//! assert!(0.is_zero());
+//! assert!(!5.is_zero());
+//! assert!(5.is_non_zero());
+//! assert!(!0.is_non_zero());
+//! ```
```
Comment (enitrat) [inline corelib/src/zeroable.cairo:18]:
```suggestion
//! Types and traits for handling non-zero values and zero checking operations.
//! 
//! This module provides the [`NonZero`] wrapper type which guarantees that a value is never 
//! zero.
//! The [`Zeroable`] trait is meant for internal use only. The public-facing equivalent is the [`Zero`] trait.
//!
//! [`Zero`]: core::num::traits::zero::Zero
```

Your example uses the Zero trait, which isn't really related to this module. The Zeroable trait is an internal-only trait, the public version of it is Zero

File: corelib/src/zeroable.cairo [hunk 2]
```diff
@@ -1,23 +1,40 @@
...
+
+// === Zero ===
```
Comment (enitrat) [inline corelib/src/zeroable.cairo:20]:
```suggestion
```

File: corelib/src/zeroable.cairo [hunk 3]
```diff
@@ -42,24 +59,19 @@ pub(crate) trait Zeroable<T> {
     fn is_non_zero(self: T) -> bool;
 }
 
-/// Provides an implementation of the `Zeroable` trait for types that implement `Zero`.
 pub(crate) mod zero_based {
-    /// Implements `Zeroable` for any type that implements `Zero`, `Drop`, and `Copy`.
```
Comment (enitrat) [inline corelib/src/zeroable.cairo:47]: I would add this back

File: corelib/src/zeroable.cairo [hunk 4]
```diff
@@ -71,9 +83,6 @@ pub(crate) impl Felt252Zeroable = zero_based::ZeroableImpl<felt252>;
 
 // === NonZero ===
 
-/// A wrapper type for non-zero values of type T.
-///
-/// This type guarantees that the wrapped value is never zero.
```
Comment (enitrat) [inline corelib/src/zeroable.cairo:76]: add this back

File: corelib/src/zeroable.cairo [hunk 5]
```diff
@@ -85,30 +94,20 @@ impl NonZeroNeg<T, +Neg<T>, +TryInto<T, NonZero<T>>> of Neg<NonZero<T>> {
...
 
-/// Represents the result of checking whether a value is zero.
 pub(crate) enum IsZeroResult<T> {
-    /// Indicates that the value is zero.
     Zero,
-    /// Indicates that the value is non-zero, wrapping it in a NonZero<T>.
     NonZero: NonZero<T>,
 }
 
-/// Unwraps a NonZero<T> to retrieve the underlying value of type T.
```
Comment (enitrat) [inline corelib/src/zeroable.cairo:96]: add this back

**Review by enitrat — CHANGES_REQUESTED** [review]

File: corelib/src/zeroable.cairo (hunk 3 above)
Comment (TAdev0) [reply corelib/src/zeroable.cairo:47]: done with `//` not to generate doc

File: corelib/src/zeroable.cairo (hunk 4 above)
Comment (TAdev0) [reply corelib/src/zeroable.cairo:76]: done

File: corelib/src/zeroable.cairo (hunk 5 above)
Comment (TAdev0) [reply corelib/src/zeroable.cairo:96]: same, added with `//` as its not supposed to be doc, it s not public

Comment (enitrat) [reply corelib/src/zeroable.cairo:96]: oh yeah right mb

**Review by enitrat — APPROVED** [review]

**Pull request closed without merging**